### ISSUE TITLE
one-to-many is broken with Angular 2

### ISSUE THREAD

**Issue by jdubois** [jhipster/generator-jhipster#4972]
In the current master, if you create a one-to-many relationship, which is not required, and then create an entity without the relationship (as it's not required), you get a:

    Cannot read property 'outlets' of null

And the UI is broken.

My configuration is below, you can test it with the `Buzz` entity.

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

```yaml
{
  "generator-jhipster": {
    "jhipsterVersion": "3.12.2",
    "baseName": "tmp",
    "packageName": "com.yolo",
    "packageFolder": "com/yolo",
    "serverPort": "8080",
    "authenticationType": "session",
    "hibernateCache": "ehcache",
    "clusteredHttpSession": false,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "buildTool": "maven",
    "enableSocialSignIn": false,
    "rememberMeKey": "cd484ea36d54bb81f90e53e414d8cc84ad4aa61b",
    "useSass": false,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": false,
    "serviceDiscoveryType": false,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "fr"
    ],
    "clientFramework": "angular2"
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

Bar.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "foo",
            "otherEntityName": "foo",
            "relationshipType": "many-to-one",
            "otherEntityField": "id"
        }
    ],
    "fields": [
        {
            "fieldName": "aaa",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170116142738",
    "dto": "no",
    "service": "serviceImpl",
    "entityTableName": "bar",
    "pagination": "pager"
}

```

Buzz.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "bar",
            "otherEntityName": "bar",
            "relationshipType": "many-to-one",
            "otherEntityField": "id"
        },
        {
            "relationshipName": "foo",
            "otherEntityName": "foo",
            "relationshipType": "many-to-many",
            "otherEntityField": "id",
            "ownerSide": true,
            "otherEntityRelationshipName": "buzz"
        }
    ],
    "fields": [
        {
            "fieldName": "zzz",
            "fieldType": "LocalDate"
        },
        {
            "fieldName": "eee",
            "fieldType": "ZonedDateTime"
        },
        {
            "fieldName": "ttt",
            "fieldType": "Youpi",
            "fieldValues": "aaa,vvv,ccc"
        },
        {
            "fieldName": "ffdd",
            "fieldType": "Boolean"
        }
    ],
    "changelogDate": "20170122144045",
    "dto": "no",
    "service": "no",
    "entityTableName": "buzz",
    "pagination": "no"
}

```

Foo.json

```yaml
{
    "fluentMethods": true,
    "relationships": [
        {
            "relationshipName": "buzz",
            "otherEntityName": "buzz",
            "relationshipType": "many-to-many",
            "ownerSide": false,
            "otherEntityRelationshipName": "foo"
        }
    ],
    "fields": [
        {
            "fieldName": "aaa",
            "fieldType": "String"
        },
        {
            "fieldName": "eee",
            "fieldType": "String"
        },
        {
            "fieldName": "fff",
            "fieldType": "String"
        },
        {
            "fieldName": "ggg",
            "fieldType": "String"
        }
    ],
    "changelogDate": "20170104093309",
    "dto": "no",
    "service": "no",
    "entityTableName": "foo",
    "pagination": "no"
}

```

##### **Browsers and Operating System**

java version "1.8.0_112"
Java(TM) SE Runtime Environment (build 1.8.0_112-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.112-b16, mixed mode)

git version 2.11.0

node: v7.4.0

npm: 4.0.5

bower: 1.8.0

gulp:
[23:32:32] CLI version 1.2.2

yeoman: 1.8.5

yarn: 0.19.1

Docker version 1.13.0, build 49bf474

docker-compose version 1.10.0, build 4bd6f1a

**Comment by VictorADS** [comment]
Should be fixed by #4948

**Comment by jdubois** [comment]
Indeed, it's fixed, thanks @VictorADS !!
